addons/portal/controllers/portal.py

```diff
@@ -321,10 +321,50 @@ def account(self, redirect=None, **post):
     def on_account_update(self, values, partner):
         pass
 
+    @route('/my/users_accesses', type='http', auth='user', website=True, methods=['GET', 'POST'])
+    def users_accesses(self, **kw):
```

Comment:
While we are adding a new route, we prefer to split get and post ones.
So the GET route will be the display of the page and the post one will only send (or resend) the invitation. The post one will redirect on the get one once it's done.
Also, in the post route, you could name your arguments (portal_user_id=None, new_user_email=None) so we avoid using kwargs, as there are only two possible arguments.

Reply:
Ok for the split between get and post, but I don't think post should redirect to get, since if there is an error in the post, it should be displayed. I instead used a system of 4 functions:
`_prepare_users_accesses_values` -> generic values for the view
`_get_commercial_partners_and_users` -> return commercial_partners and portal_users, useful to invite or reinvite a partner (and portal_users is the list of displayed users in the view)
`users_accesses_invitation` -> the POST route
`users_accesses` -> the GET route